File: DungeonCrawlers/Components/Weapon.cs
namespace DungeonCrawlers.Components
{
    public class Weapon : IWeapon
    {
        public Weapon(string name, string attackDescription, int maximumDamage, int minimumDamage)
        {
            Name = name;
            AttackDescription = attackDescription;
            MaximumDamage = maximumDamage;
            MinimumDamage = minimumDamage;
        }

        public string Name { get; }
        public string AttackDescription { get; }
        public int MaximumDamage { get; }
        public int MinimumDamage { get; }
    }
}

File: models/weapon.go
package models

type Weapon struct {
	MaximumDamage uint32
	MinimumDamage uint32
}

func createWeapon(minimumDamage uint32, maximumDamage uint32) Weapon {
	return Weapon{
		MaximumDamage: maximumDamage,
		MinimumDamage: minimumDamage,
	}
}
